hail/python/hail/ir/ir.py

```diff
@@ -187,12 +187,12 @@ def __init__(self, typ):
         self._typ = typ
 
     def _handle_randomness(self, create_uids):
-        if create_uids:
-            if isinstance(self.typ.element_type, tstruct):
-                new_elt_typ = self.typ.element_type._insert_field(uid_field_name, tint64)
-            else:
-                new_elt_typ = ttuple(tint64, self.typ.element_type)
-            return NA(tstream(new_elt_typ))
+        assert create_uids
```

Comment:
Calls to `_handle_randomness` are guarded in `IR.handle_randomness`, which should be the only caller:
```
        if (create_uids == self.has_uids) and not self.needs_randomness_handling:
            return self
        new = self._handle_randomness(create_uids)
```
so the only reason this will be called is if `create_uids` is true.